webui-src/app/mail/mail_attachment.js

```diff
@@ -1,24 +1,64 @@
 const m = require('mithril');
+const rs = require('rswebui');
 const util = require('mail/mail_util');
 
 const Layout = () => {
+  const files = [];
+  let changeView = false;
+
   return {
+    oninit: (v) => {
+      v.attrs.list.forEach(async (element) => {
+        const res = await rs.rsJsonApiRequest('/rsMsgs/getMessage', {
+          msgId: element.msgId,
+        });
+        if (res.body.retval) {
+          res.body.msg.files.forEach((element) => {
+            files.push({ ...element, from: res.body.msg.from, ts: res.body.msg.ts });
+          });
+        }
+      });
+    },
     view: (v) => [
       m('.widget', [
-        m('h3', 'Attachment'),
+        m('div.msg-attachment-container', [
+          m('h3', 'Attachment'),
+          m('.view', [
+            m(
+              '.mail-view',
+              {
+                onclick: () => (changeView = false),
+                style: { backgroundColor: changeView ? '#fff' : '#019DFF' },
+              },
+              m('i.fas.fa-mail-bulk')
+            ),
+            m(
+              '.attachment-view',
+              {
+                onclick: () => (changeView = true),
+                style: { backgroundColor: changeView ? '#019DFF' : '#fff' },
+              },
+              m('i.fas.fa-file')
+            ),
+          ]),
+        ]),
         m('hr'),
-        m(
-          util.Table,
-          m(
-            'tbody',
-            v.attrs.list.map((msg) =>
-              m(util.MessageSummary, {
-                details: msg,
-                category: 'system',
-              })
-            )
-          )
-        ),
+        changeView
+          ? m(util.AttachmentSection, {
+              files,
```

Comment:
For AttachmentSection, you always need to pass the same data for files everytime, why not fetch it inside that component itself?

Reply:
I was actually suggesting we could pass the msgId to the component so that it could fetch,

but looks like files are already fetched and being used outside of it. So fetching again is not needed.